functions/modules/stats/index.js

```diff
@@ -52,8 +55,20 @@ exports.userUpdate = functions
             updates.eventUpdateTeamCount = fieldValue.increment(1);
         }
 
+        // If user connection_history is updated, increment countConsecutiveDays or reset to 1
+        if (
+            previousValues.connection_history.length <
+            newValues.connection_history.length
+        ) {
+            const isDayInStreak = dayInStreak(newValues.connection_history);
+
+            updates.countConsecutiveDays = isDayInStreak
+                ? fieldValue.increment(1)
+                : 1;
+        }
```

Comment:
Verify the logic of this snippet : the calculation will be wrong in not conventional scenarii